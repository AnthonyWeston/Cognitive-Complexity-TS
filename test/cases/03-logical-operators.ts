function allAnd() {
    true && !true && !!true;
}

function allOr() {
    true || !true || !!true;
}

function twoSequences() {
    true && !true && !!true || !true || true;
}

function threeSequences() {
    true && !true || !!true || true && !!true;
}

function noOp() {
    true;
}
