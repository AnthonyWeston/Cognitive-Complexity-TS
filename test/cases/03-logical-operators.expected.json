{
    "03-logical-operators.ts": {
        "score": 14,
        "inner": [
            {
                "name": "allAnd",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "allOr",
                "score": 1,
                "line": 5,
                "column": 1
            },
            {
                "name": "allNullCoalescence",
                "score": 1,
                "line": 9,
                "column": 1
            },
            {
                "name": "twoSequences",
                "score": 2,
                "line": 13,
                "column": 1
            },
            {
                "name": "threeSequences",
                "score": 3,
                "line": 17,
                "column": 1
            },
            {
                "name": "noOp",
                "score": 0,
                "line": 21,
                "column": 1
            },
            {
                "name": "parenthesesDoesNotBreakSequence",
                "score": 1,
                "line": 25,
                "column": 1
            },
            {
                "name": "parenthesesAroundASequenceBreak",
                "score": 2,
                "line": 29,
                "column": 1
            },
            {
                "name": "parenthesesCanBreakSequence",
                "score": 3,
                "line": 33,
                "column": 1
            }
        ]
    }
}
